src/algorithms/peps_opt.jl

```diff
@@ -157,6 +157,13 @@ function fixedpoint(
         finalize! = (x, f, g, numiter) -> fin!(symm_finalize!(x, f, g, numiter)..., numiter)
     end
 
+    if scalartype(C) <: Real || scalartype(T) <: Real
```

Comment:
I think you can simply call `scalartype(envs)` to avoid having the where clauses.

Reply:
Of course I forgot that `CTMRGEnv` supports VectorInterface lol